tests/test_utils.py

```diff
@@ -68,7 +69,7 @@ def test_write_csv_valid_unicode(generate_data, tmpdir):
     with open(path, 'rb') as f:
```

Comment:
I highly recommend searching to make sure all cases of `open(..., 'rb')` actually get accessed as bytes, using `b""` values and whatnot.

Reply:
I think there is some incompatibility between 2 and 3 here in terms of IO behavior, but I'll take a second look.